Descent3/menu.cpp

```diff
@@ -1064,69 +1061,52 @@ bool ProcessCommandLine() {
 #define UID_MSNLB 100
 #define UID_MSNINFO 0x1000
 #define TRAINING_MISSION_NAME "Pilot Training"
-static inline int count_missions(const char *pathname, const char *wildcard) {
+
+static inline int count_missions(const std::filesystem::path &base_directory, const char *regex) {
   int c = 0;
-  char fullpath[_MAX_PATH];
-  char filename[_MAX_PATH];
-  tMissionInfo msninfo;
-  filename[0] = 0;
-  ddio_MakePath(fullpath, pathname, wildcard, NULL);
 
-  if (ddio_FindFileStart(fullpath, filename)) {
-    do {
-      const char *name;
-      ddio_MakePath(fullpath, pathname, filename, NULL);
+  ddio_DoForeachFile(base_directory, std::regex(regex), [&c](const std::filesystem::path &path) {
+    if (stricmp(path.filename().u8string().c_str(), "d3_2.mn3") == 0)
```

Comment:
`d3_2.mn3` is actually "second" part of original D3: Retribution campaign. At some point for some reason original developers just created second mn3 file instead of expanding `d3.mn3`.